specifications/unified-provider.org

```diff
@@ -28,57 +42,197 @@ Multipass) and potential future providers (Remote LXD, etc.).
 - clean_project()
 - shell()
 
-** Proposed Public Interfaces:
-
-|--------------------------------------------+---------------------------------------------------------------------------+-------------------------------------------------------|
-| method                                     | description                                                               | raises                                                |
-|--------------------------------------------+---------------------------------------------------------------------------+-------------------------------------------------------|
-| is_provider_ready() -> bool                | checks if provider is ready w/o installation or configuration             |                                                       |
-| setup_provider(echoer) -> None             | installs/configures provider, allowing for user prompting                 | ProviderInstallationError, ProviderConfigurationError |
-| execute_step(step: steps.Step) -> None     | executes (up to and including) step                                       | NotImplementedError                                   |
-| expose_prime(path: pathlib.Path) -> None   | mount project =prime= directory to host at specified path                 | NotImplementedError                                   |
-| snap() -> List[pathlib.Path]               | snap project (building as required), writing snap(s) to host project dir. |                                                       |
-| clean_parts(part_names: List[str]) -> None | clean project parts                                                       | NotImplementedError                                   |
-| clean() -> None                            | clean all project artifacts, including build-instance, if applicable.     |                                                       |
-| shell() -> None                            | launch interactive shell with build-instance, if applicable.              | NotImplementedError                                   |
-|--------------------------------------------+---------------------------------------------------------------------------+-------------------------------------------------------|
-
-*** Summary of changes:
-- Rename =ensure_provider()= to =is_provider_ready()=, scoping to see if
-  provider is ready without needing installation or further configuration.
-  Remove exceptions.  CLI will invoke setup_provider() if terminal is
-  interactive, allowing user to perform any install/configuration as necessary.
-- Remove =mount_project()=. Project path will be provided to __init__() and will
-  be verified/mounted as required.
+** Proposed draft of public interfaces
+
+#+begin_src python
+class ProviderV2:
```

Comment:
Just call it `Provider`, we won't have two versions of this living for long as we do for plugins and it is all internal anyways